components/feature/session/src/main/java/mozilla/components/feature/session/SessionIntentProcessor.kt

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session
+
+import android.content.Intent
+import android.text.TextUtils
+
+typealias IntentHandler = (Intent) -> Boolean
+
+/**
+ * Processor for intents which should trigger session-related actions.
+ */
+class SessionIntentProcessor(
+    private val sessionUseCases: SessionUseCases,
+    useDefaultHandlers: Boolean = true
+) {
+    private val defaultActionViewHandler = { intent: Intent ->
+        val url = intent.dataString
+        if (TextUtils.isEmpty(url)) {
+            false
+        }
+
+        // TODO switch to loadUrlInNewTab, once available
```

Comment:
Let's file an issue for every TODO :)

Reply:
OK: https://github.com/mozilla-mobile/android-components/issues/136